LoopKit/DeviceManager/DeviceLog/PersistentDeviceLog.swift

```diff
@@ -106,19 +108,65 @@ public class PersistentDeviceLog {
     }
 }
 
-extension NSManagedObjectContext {
+// MARK: - Simulated Core Data
+
+extension PersistentDeviceLog {
+    private var historicalEndDate: Date { Date(timeIntervalSinceNow: -.hours(24)) }
+    private var historicalEntriesPerDay: Int { 6000 }
+
+    public func generateSimulatedHistoricalDeviceLogEntries(completion: @escaping (Error?) -> Void) {
+        var startDate = Calendar.current.startOfDay(for: self.earliestLogEntryDate)
+        let endDate = Calendar.current.startOfDay(for: self.historicalEndDate)
+        var generateError: Error?
+        var entryCount = 0
+
+        self.managedObjectContext.performAndWait {
+            while startDate < endDate {
+                for index in 0..<self.historicalEntriesPerDay {
+                    let entry = DeviceLogEntry(context: self.managedObjectContext)
+                    entry.simulated(timestamp: startDate.addingTimeInterval(.hours(Double(index) * 24.0 / Double(self.historicalEntriesPerDay))))
```

Comment:
Since this is "debug only" code for us and QA it really isn't necessary that it end exactly on `endDate`. Not to mention this is only simulated data.

However, I *think* that using the same `Calendar` (or, really, the same `TimeZone`) for `Calendar.current.startOfDay` and the subsequent `Calendar.current.date(byAdding:value:to:)` already resolves the DST issues. 

There is one edge case where every DST *change* day the calculated entries performed by the inner `for` loop incorrectly uses 24 hours to distribute the entries over (rather than the correct 23 or 25 hours). I suppose I could calculate the `TimeInterval` between the current day and the next and use that in place of 24 hours, but I don't think the change is worth it considering it is debug only code.